subsys/net/lib/coap/coap.c

```diff
@@ -1236,19 +1238,7 @@ void coap_pendings_clear(struct coap_pending *pendings, size_t len)
 	}
 }
 
-static int get_observe_option(const struct coap_packet *cpkt)
-{
-	struct coap_option option = {};
-	u16_t count = 1U;
-	int r;
-
-	r = coap_find_options(cpkt, COAP_OPTION_OBSERVE, &option, count);
-	if (r <= 0) {
-		return -ENOENT;
-	}
-
-	return coap_option_value_to_int(&option);
-}
+#define get_observe_option(cpkt) get_option_int(cpkt, COAP_OPTION_OBSERVE)
```

Comment:
There is no reason to keep old function name around. It is better to rework the code to use this new public method.

If desired, it can be done as another commit where old usage is reworked..

Reply:
Added d6a145b to make change clear.